rmgpy/molecule/resonance.py

```diff
@@ -299,11 +301,28 @@ def _generate_resonance_structures(mol_list, method_list, keep_isomorphic=False,
         index += 1
 
     # check net charge
-    for mol in mol_list:
-        if mol.get_net_charge() != 0:
-            raise ResonanceError('Resonance generation gave a net charged molecule:\n{0}'
-                                 'Ions are not yet supported in RMG.'.format(
-                mol.to_adjacency_list()))
+    input_charge = mol_list[0].get_net_charge()
+
+    for mol in mol_list[1:]:
+        if mol.get_net_charge() != input_charge:
+            mol_list.remove(mol)
+            logging.warning('Resonance generation created a molecule {0} with a net charge of {1}\n'
+                            'which does not match the input mol charge of {2}'
+                            'Removing {0} from resonance structures'.format(mol.smiles,mol.get_net_charge(),input_charge))
+        if mol.contains_surface_site():
+            for x in [atom for atom in mol.atoms if atom.is_surface_site()]:
+                if x.radical_electrons != 0:
+                    mol_list.remove(mol)
+                    logging.warning('Resonance generation created a molecule {0} with {1} radicals on {2}\n'
+                    'Removing {0} from resonance structures'.format(mol.smiles,x.radical_electrons,x.symbol))
+                elif x.lone_pairs != 0:
+                    mol_list.remove(mol)
+                    logging.warning('Resonance generation created a molecule {0} with {1} lone pairs on {2}\n'
+                    'Removing {0} from resonance structures'.format(mol.smiles,x.lone_pairs,x.symbol))
+                elif x.charge != 0:
```

Comment:
I have a question about this, would it prevent all the surface species with charges?
The representative structures of N2O in gas phase are
![image](https://user-images.githubusercontent.com/56702445/119591564-d32a8f00-bda4-11eb-82ec-902a3af8c947.png)

and it might have some surface reactions that eventually contain charged species 
![image](https://user-images.githubusercontent.com/56702445/119591471-a9716800-bda4-11eb-905b-0b272cd84b53.png)

 or maybe we can just assume N2O=X, rather than a bidentate:
```
1 O u0 p2 c0 {2,D}
2 N u0 p1 c0 {1,D} {3,S}
3 N u0 p1 c0 {2,S} {4,D}
4 X u0 p0 c0 {3,D}
```

Reply:
It will not discard all surface species with formal charges. It only discards the ones that have a formal charge on the X atom like this one for example:
1 O u0 p2 c0 {2,S} {4,S}
2 N u0 p0 c+1 {1,S} {3,T}
3 N u0 p1 c0 {2,T}
4 X u0 p0 c-1 {1,S}